port/linux/omrvmem.c

```diff
@@ -964,49 +965,75 @@ default_pageSize_reserve_memory(struct OMRPortLibrary *portLibrary, void *addres
 	int flags = MAP_PRIVATE;
 	void *result = NULL;
 	int protectionFlags = PROT_NONE;
+	BOOLEAN useBackingFile = FALSE;
 
 	Trc_PRT_vmem_default_reserve_entry(address, byteAmount);
 
+	if(mode & OMRPORT_VMEM_MEMORY_MODE_FILE_HANDLE) {
+		useBackingFile = TRUE;
+	} else {
 #if defined(MAP_ANONYMOUS)
-	flags |= MAP_ANONYMOUS;
+		flags |= MAP_ANONYMOUS;
+		useBackingFile = FALSE;
 #elif defined(MAP_ANON)
-	flags |= MAP_ANON;
+		flags |= MAP_ANON;
+		useBackingFile = FALSE;
 #else
-	fd = portLibrary->file_open(portLibrary, "/dev/zero", EsOpenRead | EsOpenWrite, 0);
-	if (-1 != fd)
+		useBackingFile = TRUE;
 #endif
-	{
-		if (0 != (OMRPORT_VMEM_MEMORY_MODE_COMMIT & mode)) {
-			protectionFlags = get_protectionBits(mode);
-		} else {
-			flags |= MAP_NORESERVE;
+	}
+
+	char filename[FILE_NAME_SIZE];
+	if (useBackingFile) {
+		char int_str[FILE_NAME_SIZE / 2];
+		sprintf(int_str, "%09d", getpid()); // Max pid in 64bit system is 4194304. Pads with zeros.
+		strcpy(filename, "tempfile");
+		strcat(filename, int_str);
+
+		fd = portLibrary->file_open(portLibrary, filename, EsOpenRead | EsOpenWrite | EsOpenCreate, 0);
+		identifier->fd = fd;
+		if (-1 == fd) {
+			Trc_PRT_vmem_default_reserve_failed(address, byteAmount);
+			update_vmemIdentifier(identifier, NULL, NULL, 0, 0, 0, 0, 0, NULL);
+			Trc_PRT_vmem_default_reserve_exit(result, address, byteAmount);
+			return result;
 		}
+		ftruncate(fd, byteAmount);
+	}
 
-		/* do NOT use the MAP_FIXED flag on Linux. With this flag, Linux may return
-		 * an address that has already been reserved.
-		 */
-		result = mmap(address, (size_t)byteAmount, protectionFlags, flags, fd, 0);
+	
+	if (0 != (OMRPORT_VMEM_MEMORY_MODE_COMMIT & mode)) {
+		protectionFlags = get_protectionBits(mode);
+	} else {
+		flags |= MAP_NORESERVE;
+	}
 
-#if !defined(MAP_ANONYMOUS) && !defined(MAP_ANON)
-		portLibrary->file_close(portLibrary, fd);
-#endif
+	/* do NOT use the MAP_FIXED flag on Linux. With this flag, Linux may return
+	 * an address that has already been reserved.
+	 */
+	result = mmap(address, (size_t)byteAmount, protectionFlags, flags, fd, 0);
 
-		if (MAP_FAILED == result) {
-			result = NULL;
-		} else {
-			/* Update identifier and commit memory if required, else return reserved memory */
-			update_vmemIdentifier(identifier, result, result, byteAmount, mode, pageSize, OMRPORT_VMEM_PAGE_FLAG_NOT_USED, OMRPORT_VMEM_RESERVE_USED_MMAP, category);
-			omrmem_categories_increment_counters(category, byteAmount);
-			if (0 != (OMRPORT_VMEM_MEMORY_MODE_COMMIT & mode)) {
-				if (NULL == omrvmem_commit_memory(portLibrary, result, byteAmount, identifier)) {
-					/* If the commit fails free the memory */
-					omrvmem_free_memory(portLibrary, result, byteAmount, identifier);
-					result = NULL;
-				}
+	if(useBackingFile) {
+		portLibrary->file_close(portLibrary, fd); //Switch to unlink?
+		unlink(filename);
```

Comment:
You can `unlink()` the file from the file system right after the file is opened.   Also there is the portlibrary call for this, `portLibrary->file_unlink()`.